### DISCUSSION TITLE
How to use `useVirtualList` with a reactive list

### DISCUSSION THREAD

**Discussion by JessicaSachs** [vueuse/vueuse#824] [Q&A]
Hello! The current implementation of `useVirtualList` works great for a static list, but I'm struggling with how to make it reactive. The current signature specifies an array instead of a reactive ref. Can I get an example of `useVirtualList` with a simple textfield filter?

Here's a reproduction of what I'd like to be able to do...

https://github.com/JessicaSachs/useVirtualList/blob/main/src/App.vue#L26-L44

```js
import { ref, computed } from 'vue'
import faker from 'faker'
import { useVirtualList } from '@vueuse/core'

const search = ref('')

const bigList = Array.from(Array(100).keys()).map(a => ({
  // name, avatar, etc
  ...faker.helpers.createCard(),
  id: faker.datatype.uuid()
}))

const filteredList = computed(() => {
  return bigList.filter(i => {
    return i.name.toLocaleLowerCase().indexOf(search.value.toLocaleLowerCase()) > -1
  })
})

// Ideally, I could pass something reactive into `useVirtualList`
const { list, wrapperProps, containerProps } = useVirtualList(filteredList.value, { itemHeight: 80 })
```

![Screen Shot 2021-10-06 at 6 27 50 PM](https://user-images.githubusercontent.com/2801156/136292453-cf54465b-eb8b-4d1d-974e-cbacd38d8cb6.png)

**Comment by JMaylor** [comment]
Hey,

I had a look at this and I couldn't see a way to do it with current `useVirtualList` function, so I made some changes to make it accept a Ref[] instead of a static list.

If you check out [my fork of vueuse, on branch reactive-virtual-list](https://github.com/JMaylor/vueuse/tree/reactive-virtual-list) I made the underlying changes and updated the demo page for `useVirtualList` to do something similar to your example ^. If this looks good and one of the maintainers like the approach I can do a PR.

Thanks

**Reply by JessicaSachs** [reply]
Thank you for the example -- this helped me with the PR I created.

**Comment by JessicaSachs** [comment]
I made a PR here https://github.com/vueuse/vueuse/issues/903

**Comment by mashpie** [comment]
Alright, `useVirtualList` works fine for filtered lists. But, could it be that reordering (ie.: Array.sort()) and changes of attributes of an array element won't trigger without the `{ deep: true }` option?
